Assets/MixedRealityToolkit/_Core/Devices/MixedReality/WMRDevice.cs

```diff
@@ -0,0 +1,489 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Definitions;
+using Microsoft.MixedReality.Toolkit.Internal.Extensions;
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Internal.Managers;
+using Microsoft.MixedReality.Toolkit.Internal.Utilities;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR.WSA.Input;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Devices.WindowsMixedReality
+{
+    public class WMRDevice : IMixedRealityDevice
+    {
+        private readonly Dictionary<uint, IMixedRealityInputSource> activeControllers = new Dictionary<uint, IMixedRealityInputSource>();
+        Vector3 controllerPosition, pointerPosition, gripPosition = Vector3.zero;
+        Quaternion controllerRotation, pointerRotation, gripRotation = Quaternion.identity;
+        private IMixedRealityInputSystem inputSystem;
+
+
+        public IMixedRealityInputSource[] GetActiveControllers()
+        {
+            return activeControllers.ExportDictionaryValuesAsArray();
+        }
+
+        public WMRDevice()
+        {
+
+        }
+
+        public void Initialize()
+        {
+            inputSystem = MixedRealityManager.Instance.GetManager<IMixedRealityInputSystem>();
+
+            InitializeSources();
+        }
+
+        private void InitializeSources()
+        {
+            InteractionManager.InteractionSourceDetected += InteractionManager_InteractionSourceDetected;
```

Comment:
I'd like to use the actual `Windows.Spatial.UI` namespace to get controller data.

Reply:
Windows.UI.Input.Spatial.SpatialInteractionManager
Although I'm concerned about the "GetForCurrentView" call, as Unity may interfere.  In my experience, such a call can only work in the UWP app.

Reply:
Didn't work in the Unity project.  Although I could type and have the code recognised.  It caused errors in the Unity project.

Maybe there is some magic to have this only recognised in the Player project (and ignored in the main one?)  /cc @StephenHodgson